src/OpenTelemetry.Extensions/Trace/BaggageActivityProcessor.cs

```diff
@@ -1,21 +1,41 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
+using System;
 using System.Diagnostics;
 
 namespace OpenTelemetry.Trace;
 
 /// <summary>
-/// Activity processor that adds <see cref="Baggage"/> fields to every new span.
+/// Activity processor that adds <see cref="Baggage"/> fields to every new activity.
 /// </summary>
-internal sealed class BaggageActivityProcessor : BaseProcessor<Activity>
+public sealed class BaggageActivityProcessor : BaseProcessor<Activity>
```

Comment:
is this needed to be public?

Reply:
It needs to be public to expose the `AllowAllBaggageKeys` convenience predicate. I did originally have it in  `TracerProviderBuilderExtensions` but felt weird referencing the predicate as `TracerProviderBuilderExtensions.AllowAllBaggageKeys` and using that class directly resulted in namespace conflicts as the class name exists in a number of other packages.

Adding to `BaggageActivityProcessor` feels nicer as you then reference the predicate as `BaggageActivityProcessor.AllowAllBaggageKeys`.

I'm open alternatives, I don't like I've had to add the class and other parts to the public API.

Reply:
got it. as long as we prevent users from doing `new BaggageActivityProcessor()` (which would be incorrect), this is fine.